debezium/src/main/java/akka/stream/alpakka/debezium/impl/LazyEmbeddedEngine.java

```diff
@@ -0,0 +1,569 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.debezium.impl;
+
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.spi.OffsetCommitPolicy;
+import io.debezium.util.Clock;
+import io.debezium.util.VariableLatch;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectorContext;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.errors.RetriableException;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.runtime.distributed.DistributedConfig;
+import org.apache.kafka.connect.runtime.standalone.StandaloneConfig;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.connect.source.SourceTaskContext;
+import org.apache.kafka.connect.storage.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+public final class LazyEmbeddedEngine implements LazyDebeziumEngine<SourceRecord> {
```

Comment:
This is a adaptation of `EmbeddedEngine`. Mostly it's `EmbeddedEngine` code, but I changed how polling works: now it does not poll data by itself but rather expose a `poll` method, this way it's easy for us to control in Akka Streams.

Reply:
Also, this is just a draft. I'll try to remove more useless code later. I just want to see from you if I should go with this strategy or not.

Reply:
It certainly does make the implementation more of a burden to maintain.  Perhaps it would be beneficial to reach out to the Debezium team and ask them to review this PR. I'm sure they would be open to an upstream contribution to make life easier.